src/codec.rs

```diff
@@ -270,6 +270,75 @@ impl Encode for u64 {
     }
 }
 
```

Comment:
I mention this elsewhere, but: as a reader, I expect to see the encoder first, then the decoder.

Reply:
Whoops, I've been going alphabetically.